engine/resource/src/resource_chunk_cache.cpp

```diff
@@ -0,0 +1,445 @@
+// Copyright 2020-2024 The Defold Foundation
```

Comment:
This implementation may be used in a more generic fashion internally in the resource system, but for now our only use case is the sound streaming.